### DISCUSSION TITLE
[custom-connector] Querying a DWH as part of a connector for an API

### DISCUSSION THREAD

**Discussion by anguswalker** [airbytehq/airbyte#34287] [Connector Questions] [answered]
I am working on creating a connector for an awkward API which requires an ID per customer that expires and is aggressively rate limited. As such, I don't believe I can use the Parent Stream option as although data relating to new customers would be available expired links could not be updated. Is there a way to query a DWH (snowflake) as part of the source connector to apply logic to retrieve the list of customer IDs? I'm guessing I could just use SQLAlchemy but how can I access the credentials as its not the existing source?

**Comment by marcosmarxm** [comment, marked as answer]
You can customize the [read function](https://github.com/airbytehq/airbyte/discussions/33814) but currently no documentation for this type of change.
